modules/library/jdbc/src/main/java/org/geotools/data/jdbc/datasource/JNDIDataSourceFactory.java

```diff
@@ -43,7 +43,7 @@
 
     public static final Param JNDI_REFNAME =
             new Param(
-                    "jdniReferenceName",
+                    "jndiReferenceName",
```

Comment:
This is part of the public API so you must deprecate the misspelling and add a new correctly spelt one